packages/desktop-client/src/components/settings/Reset.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+
+import { send } from 'loot-core/src/platform/client/fetch';
+import { Text, ButtonWithLoading } from 'loot-design/src/components/common';
+
+import { ButtonSetting } from './UI';
+
+export function ResetCache() {
+  let [resetting, setResetting] = useState(false);
+
+  async function onResetCache() {
+    setResetting(true);
+    await send('reset-budget-cache');
+    setResetting(false);
+  }
+
+  return (
+    <ButtonSetting
+      button={
+        <ButtonWithLoading loading={resetting} onClick={onResetCache}>
+          Reset budget cache
+        </ButtonWithLoading>
+      }
+    >
+      <Text>
+        <strong>Reset budget cache</strong> will clear all cached values for the
+        budget and recalculate the entire budget. All values in the budget are
+        cached for performance reasons, and if there is a bug in the cache you
+        won't see correct values. There is no danger in resetting the cache.
+        Hopefully you never have to do this.
+      </Text>
+    </ButtonSetting>
+  );
+}
+
+export function ResetSync({ resetSync }) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏perhaps worth extracting this component into its own file? To have one-component/one-section per file. What do you think?

Reply:
I kept them together because they’re closely related in my eyes (like the two format options are) but happy to move them if you disagree.